_site/melville-in-london/ui/js/script.js

```diff
@@ -0,0 +1,244 @@
+var groupBasis = "group"; // this should eventually be an option defined in the Itinerary
+var excludeFromTable = ["step_type", "alt_route", "nn notes"]; // ditto
+
+function applyTemplate($templateName, $itemName, $itemContent)
+{
+    $template = $('.js-templates .'+$templateName).html();
+    $template = $template.replace('[% itemName %]', $itemName).replace('[% itemContent %]', $itemContent);
+    return $template;
+}
+
+function groupBy(array, key) {
+  return array.reduce(function(group, item) {
+    (group[item[key]] = group[item[key]] || []).push(item);
+    return group;
+  }, {});
+}
+
+$(document).ready(function() {
+  // replace with d3 color scale?
+  var colors = [
+    "#1438bd",
+    "#e9b829",
+    "#900503",
+    "#fb6b02",
+    "#038911",
+    "#281e5d",
+    "#f9a602",
+    "#d31603",
+    "#893201",
+    "#74b62d",
+    "#85144b",
+    "#141e3c",
+    "#311431"
+  ];
+
+  $.getJSON("https://mel-catalog-staging.herokuapp.com/api/itineraries/8", function(data) {
```

Comment:
Idea for future enhancement: maybe we set a MEL Catalog base URL as a config variable, which can expose any number of existing maps, to be shown in the navigation, and data and maps fetched upon load.